tests/integration/functional/test_client.py

```diff
@@ -964,3 +965,134 @@ def test_deleting_deployments(clean_client):
 
     with pytest.raises(KeyError):
         clean_client.get_deployment(str(response.id))
+
+
+class ClientCrudTestConfig(BaseModel):
+    entity_name: str
+    create_args: Dict[str, Any] = {}
+    get_args: Dict[str, Any] = {}
+    update_args: Dict[str, Any] = {}
+    delete_args: Dict[str, Any] = {}
+
+
+crud_test_configs = [
+    ClientCrudTestConfig(
+        entity_name="user",
+        create_args={"name": "user_name"},
+        update_args={"updated_name": "updated_user_name"},
+    ),
+    ClientCrudTestConfig(
+        entity_name="team",
+        create_args={"name": "team_name"},
+        update_args={"new_name": "updated_team_name"},
+    ),
+    ClientCrudTestConfig(
+        entity_name="role",
+        create_args={"name": "role_name", "permissions_list": []},
+        update_args={"new_name": "updated_role_name"},
+    ),
+    ClientCrudTestConfig(
+        entity_name="workspace",
+        create_args={"name": "workspace_name", "description": ""},
+        update_args={"new_name": "updated_workspace_name"},
+    ),
+    ClientCrudTestConfig(
+        entity_name="stack",
+        create_args={
+            "name": "stack_name",
+            "components": {
+                StackComponentType.ORCHESTRATOR: "default",
+                StackComponentType.ARTIFACT_STORE: "default",
+            },
+        },
+        update_args={"name": "updated_stack_name"},
+    ),
+    ClientCrudTestConfig(
+        entity_name="stack_component",
+        create_args={
+            "name": "stack_component_name",
+            "flavor": "local",
+            "component_type": StackComponentType.ORCHESTRATOR,
+            "configuration": {},
+        },
+        get_args={"component_type": StackComponentType.ORCHESTRATOR},
+        update_args={
+            "name": "updated_stack_component_name",
+            "component_type": StackComponentType.ORCHESTRATOR,
+        },
+        delete_args={"component_type": StackComponentType.ORCHESTRATOR},
+    ),
+    ClientCrudTestConfig(
+        entity_name="flavor",
+        create_args={
+            "source": "tests.unit.test_flavor.AriaOrchestratorFlavor",
+            "component_type": StackComponentType.ORCHESTRATOR,
+        },
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "crud_test_config",
+    crud_test_configs,
+    ids=[c.entity_name for c in crud_test_configs],
+)
+def test_basic_crud_for_entity(
+    crud_test_config: ClientCrudTestConfig, clean_client
+):
+    """Tests basic CRUD method on the client."""
+    create_method = getattr(
+        clean_client, f"create_{crud_test_config.entity_name}"
+    )
+    get_method = getattr(clean_client, f"get_{crud_test_config.entity_name}")
+    delete_method = getattr(
+        clean_client, f"delete_{crud_test_config.entity_name}"
+    )
+
+    entity = create_method(**crud_test_config.create_args)
```

Comment:
Could you please wrap this into a nice try/finally block and clean up any remaining resources at the end using the generated random name? If the test fails mid-way, those will be left on the test environment.